reana/reana_dev/run.py

```diff
@@ -538,28 +559,72 @@ def run_example(
+                    elif "running" in status:
+                        run_statistics["running"].append(workflow_name)
+                    elif "failed" in status:
+                        run_statistics["failed"].append(workflow_name)
+                    elif "finished" in status or "stopped" in status:
+
+                        if not _verify_log_output(component, workflow_name):
+                            run_statistics["failed"].append(workflow_name)
+                            continue
+
+                        missing_files = _return_missing_output_files(
+                            component, workflow_name
                         )
-                        run_command(cmd, component)
-                    # verify output file presence
-                    cmd = "reana-client ls -w {0}".format(workflow_name)
-                    listing = run_command(cmd, component, return_output=True)
-                    click.secho(listing)
-                    expected_files = file or get_expected_output_filenames_for_example(
-                        component
-                    )
-                    for expected_file in expected_files:
-                        if expected_file not in listing:
-                            click.secho(
-                                "[ERROR] Expected output file {0} not found. "
-                                "Exiting.".format(expected_file)
-                            )
-                            sys.exit(1)
-    # report that everything was OK
-    run_command("echo OK", component)
+
+                        if len(missing_files):
+                            run_statistics["failed"].append(workflow_name)
+                        else:
+                            run_statistics["passed"].append(workflow_name)
+
+    if not submit_only:
+        exit_status = 0
+        exit_message = "OK"
+
+        any_running_left = any(
+            [
+                run_statistics["running"],
+                run_statistics["pending"],
+                run_statistics["queued"],
+            ]
+        )
+        if any_running_left:
+            exit_status = 1
+            exit_message = "PROGRESS"
+
+        failed_report_message = ""
+        if run_statistics["failed"]:
+            failed_report_message = f"({', '.join(run_statistics['failed'])})"
+            exit_message = "FAILED"
+            exit_status = 2
+
+        try:
+            original_command = f"{sys.argv[0].split('/')[-1]} {' '.join(sys.argv[1:])}"
+        except IndexError:
+            original_command = ""
+
+        report_message = (
+            f"{original_command}\n"
+            "Summary:\n"
+            f"  - {len(run_statistics['running'])} running, {len(run_statistics['pending'])} pending, {len(run_statistics['queued'])} queued\n"
+            f"  - {len(run_statistics['passed'])} passed\n"
+            f"  - {len(run_statistics['failed'])} failed {failed_report_message}\n"
+            f"{exit_message}"
```

Comment:
Cosmetics: Currently we end with:

```
[2022-04-27T11:05:38] : reana-dev run-example -b kubernetes -c r-d-r-roofit --check-only
Summary:
  - 0 running, 0 pending, 0 queued
  - 4 passed
  - 0 failed
OK
```

We could end with:

```
[2022-04-27T11:05:38] reana: reana-dev run-example -b kubernetes -c r-d-r-roofit --check-only
Test matrix:
  - Demo examples: reana-demo-root6-roofit
  - Workflow engines: cwl, serial, snakemake, yadage
  - Compute backends: kubernetes
Test results:
  - 0 running, 0 pending, 0 queued
  - 4 passed
  - 0 failed
OK
```

(i.e. repeat the test matrix header for convenience, since the output can be rather long; etc)